crates/shadowsocks-service/src/local/tun/tcp.rs

```diff
@@ -203,10 +185,7 @@ impl AsyncWrite for TcpConnection {
 
         let n = control.send_buffer.enqueue_slice(buf);
 
-        if n > 0 {
-            self.manager_notify.notify();
-        }
-        Ok(n).into()
+        if n > 0 { Ok(n).into() } else { Poll::Pending }
```

Comment:
After data was copied into the `send_buffer`, we should wake up the `poll` thread to copy data from `send_buffer` into socket's send buffer, otherwise delay would be increased.